Author: Leigh Scott <leigh123linux@fedoraproject.org>
Description: Match edit dialog does not return result
 In any plugin which needs some window match rules, click + to add, grab
 some window, but the resulting match is not written to the match textfield.
Origin: https://bugzilla.redhat.com/show_bug.cgi?id=589677
--- ccsm-0.8.4.orig/ccm/Widgets.py	2009-10-14 02:05:30.000000000 +0100
+++ ccsm-0.8.4/ccm/Widgets.py	2010-05-07 16:58:37.000000000 +0100
@@ -1193,7 +1193,7 @@ class MatchButton(gtk.Button):
         dlg.show_all ()
 
         response = dlg.run ()
-        dlg.destroy ()
+        dlg.hide ()
         if response == gtk.RESPONSE_OK:
             type     = type_chooser.get_active_text ()
             value    = entry.get_text ()
@@ -1201,6 +1201,8 @@ class MatchButton(gtk.Button):
             invert   = check.get_active ()
             self.generate_match (type, value, relation, invert)
 
+        dlg.destroy ()
+
 class FileButton (gtk.Button):
     __gsignals__    = {"changed" : (gobject.SIGNAL_RUN_FIRST,
                                     gobject.TYPE_NONE,
